src/components/calcite-time-picker/calcite-time-picker.tsx

```diff
@@ -570,47 +570,90 @@ export class CalciteTimePicker {
+      this.localizedMinute = localizedMinute;
+      this.localizedMinuteSuffix = localizedMinuteSuffix;
+      this.localizedSecond = localizedSecond;
+      this.localizedSecondSuffix = localizedSecondSuffix;
+      this.hour = hour;
+      this.minute = minute;
+      this.second = second;
+      if (localizedMeridiem) {
+        this.localizedMeridiem = localizedMeridiem;
+        this.meridiem = getMeridiem(this.hour);
+      }
+    } else {
+      this.hour = null;
+      this.localizedHour = null;
+      this.localizedHourSuffix = null;
+      this.localizedMeridiem = null;
+      this.localizedMinute = null;
+      this.localizedMinuteSuffix = null;
+      this.localizedSecond = null;
+      this.localizedSecondSuffix = null;
+      this.meridiem = null;
+      this.minute = null;
+      this.second = null;
+      this.value = null;
+    }
+    if (emit) {
+      this.calciteTimePickerChange.emit(this.value);
+    }
+  };
+
+  private setValuePart = (
     key: "hour" | "minute" | "second" | "meridiem",
     value: number | string | Meridiem,
     emit = true
   ): void => {
-    switch (key) {
-      default:
-        return;
-      case "hour":
-        this.hour = typeof value === "number" ? formatTimePart(value) : value;
-        break;
-      case "minute":
-        this.minute = typeof value === "number" ? formatTimePart(value) : value;
-        break;
-      case "second":
-        this.second = typeof value === "number" ? formatTimePart(value) : value;
-        break;
-      case "meridiem":
-        if (isValidNumber(this.hour)) {
-          const hourAsNumber = parseInt(this.hour);
-          switch (value) {
-            case "AM":
-              if (hourAsNumber >= 12) {
-                this.hour = formatTimePart(hourAsNumber - 12);
-              }
-              break;
-            case "PM":
-              if (hourAsNumber < 12) {
-                this.hour = formatTimePart(hourAsNumber + 12);
-              }
-              break;
-          }
-          this.meridiem = value as Meridiem;
-        } else {
-          this.meridiem = value as Meridiem;
+    if (key === "meridiem") {
+      this.meridiem = value as Meridiem;
+      if (isValidNumber(this.hour)) {
+        const hourAsNumber = parseInt(this.hour);
+        switch (value) {
+          case "AM":
+            if (hourAsNumber >= 12) {
+              this.hour = formatTimePart(hourAsNumber - 12);
+            }
+            break;
+          case "PM":
+            if (hourAsNumber < 12) {
+              this.hour = formatTimePart(hourAsNumber + 12);
+            }
+            break;
         }
-        break;
+        this.localizedHour = localizeTimePart(this.hour, "hour", this.locale);
+      }
+    } else {
+      this[key] = typeof value === "number" ? formatTimePart(value) : value;
+      this[`localized${capitalize(key)}`] = localizeTimePart(this[key], key, this.locale);
+    }
+    if (this.hour && this.minute) {
+      this.value =
```

Comment:
Nitpick: to DRY this up a bit you could move the conditional that affects seconds within its corresponding expression:

```ts
const showSeconds = this.second && this.showSecond;
this.value = `${this.hour}:${this.minute}:${showSeconds ? this.second : "00"}`;
```